qiskit/synthesis/evolution/product_formula.py

```diff
@@ -151,7 +151,7 @@ def _single_qubit_evolution(pauli, time):
 
 def _two_qubit_evolution(pauli, time, cx_structure):
     # get the Paulis and the qubits they act on
-    labels_as_array = np.array(list(pauli.to_label()))
+    labels_as_array = np.array(list(reversed(pauli.to_label())))
```

Comment:
It wouldn't hurt to put in a one-line comment saying that any phase should be handled safely by being pushed outside the indices of allowed qubits - as it stands, we might come back to it in a few months and think it's not being handled.

Reply:
done in 9c5e256e4